### PR TITLE
Refactor demo project

### PR DESCRIPTION
**Pull request by hoangchungk53qx1**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #[issue number]

**Description**
refactor demo project. 

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: (Bug fix | Feature | Documentation | Testing | Code health | Builds | Releases | Other)

**Screenshots (if applicable)**

**Checklist**
- [v] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/CODE_OF_CONDUCT.md).
- [v] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [v] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [v] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [v] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [v] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [v] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

**Comment by hoangchungk53qx1** [comment]
@vorburger

**Comment by hoangchungk53qx1** [comment]
@santosh-pingle 🙏

**Comment by hoangchungk53qx1** [comment]
@jingtang10  👆

File: demo/src/main/java/com/google/android/fhir/demo/extensions/LifeCycleExt.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+package com.google.android.fhir.demo.extensions
+
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.lifecycleScope
+import androidx.lifecycle.repeatOnLifecycle
+import kotlinx.coroutines.launch
+
+fun Fragment.launchAndRepeatStarted(
```
Comment (MJ1998) [inline demo/src/main/java/com/google/android/fhir/demo/extensions/LifeCycleExt.kt:25]: What is this trying to do ?

**Review by MJ1998 — COMMENTED** [review]
@hoangchungk53qx1 
Thanks for your first PR.  Looking forward for more PRs from you.

Why do you wanna split the pollState into periodic and oneTime ?

File: demo/src/main/java/com/google/android/fhir/demo/extensions/LifeCycleExt.kt (hunk 1 above)
Comment (hoangchungk53qx1) [reply demo/src/main/java/com/google/android/fhir/demo/extensions/LifeCycleExt.kt:25]: A safer way to collect flows from Android UIs
https://medium.com/androiddevelopers/a-safer-way-to-collect-flows-from-android-uis-23080b1f8bda

Comment (hoangchungk53qx1) [reply demo/src/main/java/com/google/android/fhir/demo/extensions/LifeCycleExt.kt:25]: I want to split the 2 states into 2 different hot flows, and each time the periodic trigger will be activated.

Comment (hoangchungk53qx1) [reply demo/src/main/java/com/google/android/fhir/demo/extensions/LifeCycleExt.kt:25]: > What is this trying to do ?
A safer way to collect flows from Android UIs
https://medium.com/androiddevelopers/a-safer-way-to-collect-flows-from-android-uis-23080b1f8bda

**Comment by hoc081098** [comment]
Looks good🙏 @hoangchungk53qx1

**Review by jingtang10 — APPROVED** [review]
thanks chung for the contribution. i think this change does make sense and will approve now.

in the future, can you please try to use more descriptive names for prs and perhaps send more specific changes (for example in this case splitting the flow into 2 can be a separate pr, and adding the file utility can be another).

but all in all very good. thanks for the pr again! look forward to receiving more contributions from you.

**Comment by hoangchungk53qx1** [comment]
> thanks chung for the contribution. i think this change does make sense and will approve now.
> 
> in the future, can you please try to use more descriptive names for prs and perhaps send more specific changes (for example in this case splitting the flow into 2 can be a separate pr, and adding the file utility can be another).
> 
> but all in all very good. thanks for the pr again! look forward to receiving more contributions from you.

I just made a few changes to this PR, thank you !

**Comment by hoc081098** [comment]
@hoangchungk53qx1 🙏

**Pull request merged**